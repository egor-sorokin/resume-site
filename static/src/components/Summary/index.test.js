// eslint-disable-next-line 
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import Summary from './index';


describe('<Summary />', () => {
  let component;
  const summary = [
    {
      title: "test title",
      text: "test text"
    },
    {
      title: "test title 2",
      text: "test text 2"
    }
  ];


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<Summary />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 0 component', () => {
      expect(component.children()).toHaveLength(0);
    });
  });


  describe('props tests', () => {
    it('renders summary items', () => {
      component = shallow(<Summary summary={summary}/>);

      expect(component.find('li')).toHaveLength(2);
      expect(component.find('li').at(0).find('.summary__title').text()).toEqual(summary[0].title);
      expect(component.find('li').at(1).find('.summary__title').text()).toEqual(summary[1].title);
    });
  });
}); 
