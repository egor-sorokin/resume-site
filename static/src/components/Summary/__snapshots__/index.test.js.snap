// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> base tests matches snapshot 1`] = `
<ul
  className="summary"
/>
`;
